src/veins/modules/utility/BBoxLookup.cc

```diff
@@ -111,12 +111,16 @@ BBoxLookup::BBoxLookup(const std::vector<Obstacle*>& obstacles, std::function<BB
     size_t numEntries = 0;
     for (const auto obstaclePtr : obstacles) {
         auto bbox = makeBBox(obstaclePtr);
-        const size_t fromCol = std::max(0, int(bbox.p1.x / cellSize));
-        const size_t toCol = std::max(0, int(bbox.p2.x / cellSize));
-        const size_t fromRow = std::max(0, int(bbox.p1.y / cellSize));
-        const size_t toRow = std::max(0, int(bbox.p2.y / cellSize));
+        const size_t fromCol = std::min(size_t(std::max(0, int(bbox.p1.x / cellSize))), numCols - 1);
+        const size_t toCol = std::min(size_t(std::max(0, int(bbox.p2.x / cellSize))), numCols - 1);
+        const size_t fromRow = std::min(size_t(std::max(0, int(bbox.p1.y / cellSize))), numRows - 1);
+        const size_t toRow = std::min(size_t(std::max(0, int(bbox.p2.y / cellSize))), numRows - 1);
```

Comment:
```suggestion
        const size_t fromCol = std::min(size_t(std::max(0, size_t(bbox.p1.x / cellSize))), numCols - 1);
        const size_t toCol = std::min(size_t(std::max(0, size_t(bbox.p2.x / cellSize))), numCols - 1);
        const size_t fromRow = std::min(size_t(std::max(0, size_t(bbox.p1.y / cellSize))), numRows - 1);
        const size_t toRow = std::min(size_t(std::max(0, size_t(bbox.p2.y / cellSize))), numRows - 1);
```
Consistently use `size_t` as a type to convert to here (fix from previously used `int`)